app/controllers/concerns/authentication.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+# Concern for handling authentication.
+# Note that this concern is based on the generated code.
+module Authentication
+  extend ActiveSupport::Concern
+
+  # Apache is configured so that:
+  # - /webauth/login requires a shibboleth authenticated user. Thus, redirecting a user to it triggers login.
+  # - /Shibboleth.sso/Logout logs the user out of shibboleth.
+  # - /queues is limited to members of the sdr:developer group.
+  # - Other pages do not require authentication.
+  # - It provides the following request headers:
+  #  - X-Remote-User
+  #  - X-Groups
+  #  - X-Person-Name (first name)
+  #  - X-Person-Formal-Name (full name)
```

Comment:
I believe this is a thing only for H2 and H3, where Apache is proxying standalone passenger. I'm not sure if we'll want Heracles to be set up the same way. Depends on if we need websockets or not.